### PR TITLE
Change Maven wrapper settings

### PR DESCRIPTION
**Pull request by seanf**

### CODE REVIEW

**Comment by ikysil** [comment]
Dear Sean,

Could you please drop a couple of lines describing the purpose of this change?

**Comment by seanf** [comment]
Okay, it's been a while, let me see...

Well, Maven Wrapper allows us to control the version of Maven used for compiling the project (instead of picking up the random version of Maven you may or may not have on your PATH). Maven Wrapper installs Maven to a directory outside the source tree (normally under `~/.m2/wrapper`) which is expected to be reused between builds. This change is about letting Travis cache those files for faster builds.

Also, Maven Wrapper's script `mvnw` needs the file `maven-wrapper.jar` in the source tree to do its work. Maven Wrapper _can_ download that jar if required, but part of this change is about caching _that_ file under `~/.m2/wrapper` also, and then putting it back in the source tree where `mvnw` can use it. 

If it still works as it should, Maven Wrapper should not need to download anything once the cache has been populated.

File: .travis.yml [hunk 1]
```diff
@@ -68,6 +70,8 @@ matrix:
 script:
 - |
   set -e
+  # copy wrapper jar from cache, if it exists, but don't overwrite one in the source tree, if it exists:
+  false | cp -i $HOME/.m2/wrapper/maven-wrapper.jar $TRAVIS_BUILD_DIR/.mvn/wrapper/ 2>/dev/null || true
```
Comment (seanf) [inline .travis.yml]: Looking at this code now, I think this was a shell trick to ensure that the whole line was considered successful even if the `cp` failed. There's probably a better option.

**Comment by ikysil** [comment]
Closing this PR **WITHOUT MERGE** - see #188

**Pull request closed without merging**